### PR TITLE
472 wrong team

### PR DESCRIPTION
**Pull request by teodosii**
Fix for #472

### CODE REVIEW

File: engine/apps/api/views/user.py [hunk 1]
```diff
@@ -245,6 +249,26 @@ def retrieve(self, request, *args, **kwargs):
         serializer = self.get_serializer(instance, context=context)
         return Response(serializer.data)
 
+    def wrong_team_response(self):
+        """
+        This method returns 403 and {"error_code": "wrong_team", "owner_team": {"name", "id", "email", "avatar_url"}}.
```
Comment (vadimkerr) [inline engine/apps/api/views/user.py:254]: probably worth adding a comment why TeamFilteringMixin cannot be used here

**Review by vadimkerr — APPROVED** [review]

File: grafana-plugin/src/models/base_store.ts [hunk 2]
```diff
@@ -37,10 +41,13 @@ export default class BaseStore {
   }
 
   @action
-  async getById(id: string) {
-    return await makeRequest(`${this.path}${id}/`, {
+  async getById(id: string, skipErrorHandling = false, fromOrganization = false) {
+    let path = `${this.path}${id}`;
+    if (fromOrganization) {path = path.concat('?from_organization=true');}
+
+    return await makeRequest(path, {
```
Comment (maskin25) [inline grafana-plugin/src/models/base_store.ts:48]: `params: { from_organization:fromOrganization? true: undefined}`

File: grafana-plugin/src/models/schedule/schedule.ts [hunk 3]
```diff
@@ -25,6 +25,18 @@ export class ScheduleStore extends BaseStore {
     this.path = '/schedules/';
   }
 
+  @action
+  async loadItem(id: Schedule['id'], skipErrorHandling = false): Promise<Schedule> {
```
Comment (maskin25) [inline grafana-plugin/src/models/schedule/schedule.ts:29]: great that you add typings for promises!

**Comment by maskin25** [comment]
I don't like that you copypasted 

```
notFound?: boolean;
  wrongTeamError?: boolean;
  teamToSwitch?: { name: string; id: string };
  wrongTeamNoPermissions?: boolean;
```
to many pages, it would be good to wrap this state, requests, logic and so on into separate service/top level component and manage this stuff in one place to avoid copypaste

**Review by maskin25 — CHANGES_REQUESTED** [review]

**Review by maskin25 — APPROVED** [review]

**Pull request merged**